packages/lib/slots.ts

```diff
@@ -184,29 +206,28 @@ function buildSlotsWithDateRanges({
     }
   }
 
+  const startTimeWithMinNotice = dayjs.utc().add(minimumBookingNotice, "minute");
+
+  // to stay consistent with past behaviour we need to adjust the dateRanges according to frequency.
+  // this prevents slots appearing as 11:00, 11:15, 11:45, 12:00, .. etc.
+  adjustDateRanges(dateRanges, frequency);
+
   dateRanges.forEach((range) => {
     const dateYYYYMMDD = range.start.format("YYYY-MM-DD");
-    const startTimeWithMinNotice = dayjs.utc().add(minimumBookingNotice, "minute");
 
     let slotStartTime = range.start.utc().isAfter(startTimeWithMinNotice)
       ? range.start
       : startTimeWithMinNotice;
 
+    // strictly speaking not required anymore, but we keep it for now.
     slotStartTime =
       slotStartTime.minute() % interval !== 0
         ? slotStartTime.startOf("hour").add(Math.ceil(slotStartTime.minute() / interval) * interval, "minute")
         : slotStartTime;
 
-    // Adding 1 minute to date ranges that end at midnight to ensure that the last slot is included
```

Comment:
Removing organizerTimeZone, the way it was passed was highly buggy as we couldn't be sure if it was the right timeZone at this point in time. Instead moving it to date-ranges, where I'll round up to the interval.